src/Microsoft.ML/MemoryCollection.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML
+{
+    public class MemoryCollection
+    {
+        /// <summary>
+        /// Creates memory collection loader.
+        /// </summary>
+        public static MemoryCollectionLoader<T> Create<T>(IList<T> data) where T:class
+        {
+            return new MemoryCollectionLoader<T>(data);
+        }
+
+        /// <summary>
+        /// Creates memory collection loader.
+        /// </summary>
+        public static MemoryCollectionLoader<T> Create<T>(IEnumerable<T> data) where T : class
+        {
+            return new MemoryCollectionLoader<T>(data);
+        }
+    }
+
+    /// <summary>
+    /// Allows you to convert your memory collection into IDataview.
+    /// </summary>
+    /// <typeparam name="TInput"></typeparam>
+    public class MemoryCollectionLoader<TInput> : ILearningPipelineLoader
+        where TInput : class
+    {
+        public ILearningPipelineStep ApplyStep(ILearningPipelineStep previousStep, Experiment experiment)
+        {
+            Contracts.Assert(previousStep == null);
+            _dataViewEntryPoint = new Data.DataViewReference();
+            var importOutput = experiment.Add(_dataViewEntryPoint);
+            return new MemoryCollectionPipelineStep(importOutput.Data);
+        }
+
+        private readonly IList<TInput> _listCollection;
+        private readonly IEnumerable<TInput> _enumerableCollection;
+
+        private Data.DataViewReference _dataViewEntryPoint;
+        private IDataView _dataView;
+
+        /// <summary>
+        /// Creates IDataview on top of collection
+        /// </summary>
+        public MemoryCollectionLoader(IList<TInput> collection)
+        {
+            Contracts.CheckParamValue(Utils.Size(collection) > 0, collection, nameof(collection), "Must be non-empty");
+            _listCollection = collection;
+        }
+
+        /// <summary>
+        /// Creates IDataview on top of collection
+        /// </summary>
+        public MemoryCollectionLoader(IEnumerable<TInput> collection)
+        {
+            Contracts.CheckValue(collection,nameof(collection), "Must be non-null");
```

Comment:
>, "Must be non-null" [](start = 62, length = 20)

Remove this string literal, it is not necessary given that you're using this check. The messages for all these are standardized, and it's better for them to be consitent.